qlty-check/src/executor.rs

```diff
@@ -774,20 +721,28 @@ fn run_invocation(
     Ok(result)
 }
 
-fn walk_collect_entries_parallel(builder: &WalkBuilder) -> Vec<DirEntry> {
-    let dents = Arc::new(Mutex::new(vec![]));
+fn ensure_parent_exists(path: &Path) -> Result<()> {
+    if let Some(parent) = path.parent() {
+        if !parent.exists() {
+            std::fs::create_dir_all(parent)
+                .with_context(|| format!("Failed to create directory: {}", parent.display()))?;
+        }
+    } else {
+        error!(
+            "No parent directory for destination file: {:?}, this could cause issues",
+            path
+        );
+    }
 
-    builder.build_parallel().run(|| {
-        let dents = dents.clone();
+    Ok(())
+}
 
-        Box::new(move |result| {
-            if let Ok(dent) = result {
-                dents.lock().unwrap().push(dent);
-            }
-            WalkState::Continue
-        })
-    });
+#[cfg(unix)]
+fn create_symlink(from: &Path, to: &Path) -> std::io::Result<()> {
+    std::os::unix::fs::symlink(from, to)
+}
```

Comment:
It feels like the symlink stuff and `ensure_parent_exists` is ripe to be isolated into a file like `fs.rs` somewhere where it is reusable and out of the way